src/rules/__tests__/no-confusing-set-timeout.test.ts

```diff
@@ -0,0 +1,172 @@
+import { TSESLint } from '@typescript-eslint/utils';
+import dedent from 'dedent';
+import rule from '../no-confusing-set-timeout';
+import { espreeParser } from './test-utils';
+
+const ruleTester = new TSESLint.RuleTester({
+  parser: espreeParser,
+  parserOptions: {
+    ecmaVersion: 2020,
+  },
+});
+
+ruleTester.run('no-confusing-set-timeout', rule, {
```

Comment:
These look like a good start, but it would be good to also have some more granular ones to make it easier to narrow down problems, i.e. single line calls to `setTimeout`, `window.setTimeout`, etc, a case with literally just two `jest.setTimeout` calls, and so on.